### ISSUE TITLE
test whether the summaries are correlated or not!

### ISSUE THREAD

**Issue by mjvakili** [ErikOsinga/master_research_project#8]
Can you look at the output of the initial IMNN training and see whether {xi} parameters are correlated or not?

**Comment by ErikOsinga** [comment]
Yes, they seem to be correlated

The correlation matrix is the following:
`[[ 1.        -0.2900731]
 [-0.2900731  1.       ]]`

![x1vsx2](https://user-images.githubusercontent.com/12582441/57840978-f68bce00-77c9-11e9-8d6c-7196626bf1c5.png)

**Comment by mjvakili** [comment]
But didn't these points correspond to different cosmologies?

**Comment by ErikOsinga** [comment]
Ah yes. When I put 5000 noisy datavectors generated with the fiducial parameters into the network the output is as follows

![output_fid](https://user-images.githubusercontent.com/12582441/57846554-c990e880-77d4-11e9-93f1-b89bb031824b.png)

The output summaries are thus highly correlated:

```
[[1.         0.95541443]
 [0.95541443 1.        ]]
```

**Comment by mjvakili** [comment]
Nice! does that mean that the additional regularization function added to the objective function promotes perfect correlation instead of zero correlation? Can you check the code to see how the regularization look like?

Does this also imply that in order to constrain the parameters you can safely only use x1?

**Comment by ErikOsinga** [comment]
The regularization function seems to be implemented correctly,
`        square_norm = tf.reduce_sum(
            tf.square(
                tf.subtract(
                    cov,
                    tf.eye(self.n_summaries))),
            name="square_norm_covariance")
        coupling = tf.placeholder(
            dtype=self._FLOATX,
            shape=(),
            name="coupling")
        loss = tf.subtract(
            tf.multiply(coupling, square_norm), logdetfisher, name="loss")`

I think since \Omega_m and \sigma_8 are degenerate, it is possible to constrain the combination of the two parameters with only x1.

**Comment by mjvakili** [comment]
Yes. This seems reasonable. Perhaps switching the parameters to to S8 and Omegam will change it. 
See issue #10

**Comment by ErikOsinga** [comment]
Summaries are still correlated for S8 and Omega_m:

![summaries_3](https://user-images.githubusercontent.com/12582441/58865794-7addbc80-86b7-11e9-963e-74afe87e603c.png)
